SDK v2 migration guide.md

```diff
@@ -12,8 +12,8 @@ are outlines of the notable breaking changes as well as a mapping from version 1
    - [IoT hub device client](#iot-hub-device-client)
    - [IoT hub service client](#iot-hub-service-client)
    - [Device Provisioning Service (DPS) device client](#dps-device-client)
-   - [DPS service slient](#dps-service-client)
-   - [Security provider clients](#security-provider-clients)
```

Comment:
We use the old terms and point them to the new ones, so this old line was correct.

Reply:
Sure, I will make sure to update the broken link only